### ISSUE TITLE
after:run isn't triggered when closing the project after `cypress open`

### ISSUE THREAD

**Issue by TheDutchCoder** [cypress-io/cypress#17532]
### Current behavior

It looks like `after:run` never triggers.
```ts
module.exports = async (on: any, config: { env: { awsConfig: any } }) => {
  // AWS config
  config.env.awsConfig = awsconfig

  on('after:run', () => {
    console.log('cleaning')
    clean()
  })
}
```

I'm not sure how it's supposed to function, the docs say it should trigger after closing the project (after `cypress open`), but it doesn't do that.

Maybe these docs are incomplete? It also looks like there's at least a typo here: "You can return a promise from the after:run event handler and it will be awaited **before Cypress proceeds running your specs**." (I'm sure that means to say **after**?)

### Desired behavior

This trigger to be called.

### Test code to reproduce

See my above piece of code.

### Cypress Version

8.0.0

**Comment by jennifer-shehane** [comment]
@TheDutchCoder How are you closing the project via `cypress open`?

**Comment by TheDutchCoder** [comment]
> @TheDutchCoder How are you closing the project via `cypress open`?

I just close the window. I don't think there's another way?

**Comment by robert-elles** [comment]
I have the same problem. after:run event is not run when you are using the test runner tool. When you are finished running tests you will close the tool window. At that point I expect cypress to trigger the after:run event.

**Comment by robert-elles** [comment]
@jennifer-shehane I think the information has been provided

**Comment by robert-elles** [comment]
This links to issue #18955

**Comment by TheDutchCoder** [comment]
Is there a reason this issue still is in this state?
[I've seen a newer opened issue ](https://github.com/cypress-io/cypress/issues/18955) is the same issue and confirmed.

It is still a real problem, as our DB/user seeds don't get cleaned up currently because of it, making writing tests incredibly hard.

**Comment by TheDutchCoder** [comment]
I guess this is where this functionality is triggered: https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/project-base.ts#L343

Since none of this is try/catched, I think something throws and it never hits the hook at the end?

It's probably a good idea to do a try/catch/finally with the hook running in the finally clause?

**Comment by dexcell** [comment]
same issue here!

**Comment by alexniculae** [comment]
Did anybody find a workaround for this until the issue is addressed and fixed?

**Comment by Alm0g** [comment]
Any news regarding this issue?

**Comment by xavbourdeau** [comment]
Same issue here. Any workaround ?

**Comment by xavbourdeau** [comment]
Go the solution:  Set `experimentalInteractiveRunEvents` flag to `true` (see [documentation](https://docs.cypress.io/guides/references/experiments))

> Allows listening to the [before:run](https://docs.cypress.io/api/plugins/before-run-api), [after:run](https://docs.cypress.io/api/plugins/after-run-api), [before:spec](https://docs.cypress.io/api/plugins/before-spec-api), and [after:spec](https://docs.cypress.io/api/plugins/after-spec-api) events in the [setupNodeEvents](https://docs.cypress.io/guides/tooling/plugins-guide#Using-a-plugin) function during interactive mode.

**Comment by sutapaGanguly** [comment]
Setting experimentalInteractiveRunEvents flag to true is also not triggering the after:run.

Tried closing the browser but no luck with the same. after:spec works fine. There is issue with after:run IMO.

**Comment by patrickcate** [comment]
Ran into this as well. It is causing issues with data cleanup.

**Comment by funador** [comment]
Same as everyone else, want to tear some data down post-run but can't :/ If anyone has a workaround, please post it up 🙏

**Comment by matthiasreinartz** [comment]
Did anybody find a workaround? This bug still seems to exist even in the latest Cypress version

**Comment by lilywang153** [comment]
same issue here with the latest Cypress version

**Comment by jennifer-shehane** [comment]
Duplicate of https://github.com/cypress-io/cypress/issues/18955
